src/mapnik_image.cpp

```diff
@@ -1620,6 +1644,131 @@ void Image::EIO_AfterOpen(uv_work_t* req)
     delete closure;
 }
 
+// Read from a Buffer
+NAN_METHOD(Image::fromSVGBytesSync)
+{
+    NanScope();
+    NanReturnValue(_fromSVGSync(false, args));
+}
+
+// Read from a file
+NAN_METHOD(Image::fromSVGSync)
+{
+    NanScope();
+    NanReturnValue(_fromSVGSync(true, args));
+}
+
+Local<Value> Image::_fromSVGSync(bool fromFile, _NAN_METHOD_ARGS)
+{
+    NanEscapableScope();
+
+    if (!fromFile && (args.Length() < 1 || !args[0]->IsObject())) {
+        NanThrowTypeError("must provide a buffer argument");
+        return NanEscapeScope(NanUndefined());
+    }
+
+    if (fromFile && (args.Length() < 1 || !args[0]->IsString())) {
+        NanThrowTypeError("must provide a filename argument");
+        return NanEscapeScope(NanUndefined());
+    }
+
+
+    double scale = 1.0;
+    if (args.Length() >= 2) {
+        if (!args[1]->IsObject()) {
+            NanThrowTypeError("optional second arg must be an options object");
+            return NanEscapeScope(NanUndefined());
+        }
+        Local<Object> options = args[1]->ToObject();
+        if (options->Has(NanNew("scale")))
+        {
+            Local<Value> scale_opt = options->Get(NanNew("scale"));
```

Comment:
Don't see a unit test to cover this error section, and error should be raised if scale is negative or zero.